tests/jobs/ahm_analysis/test_integration.py

```diff
@@ -0,0 +1,102 @@
+import os
+import shutil
+import pytest
+import pandas as pd
+import semeio.workflows.ahm_analysis.ahmanalysis as ahmanalysis
+
+
+from res.enkf import EnKFMain, ResConfig
+from res.enkf.export import (
+    SummaryObservationCollector,
+    GenDataObservationCollector,
+)
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_ahmanalysis_run(test_data_root):
```

Comment:
Unless I'm mistaken, the only tests for the `run` method - which does include quite a bit of logic, are these integration tests. I would suggest that we assert a bit more. For instance, we could verify that an output has been created for each of the observation groups. I'm a bit split on how much we should assert of the created values, as we are here testing the job, and not the implementation in libres. But it wouldn't be outrageous to check against an expected output set.

Reply:
I have added some more tests to check all obs files are created